### PR TITLE
ADEN-9578 Add NoBid bidder

### PR DESCRIPTION
**Pull request by nandy-andy**

### CODE REVIEW

File: src/ad-engine/services/pbjs-factory.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { logger, scriptLoader } from '../utils';
 import { context } from './context-service';
 
 const prebidLibraryUrl =
-	'//origin-images.wikia.com/fandom-ae-assets/prebid.js/v2.44.0/20191213.min.js';
+	'//origin-images.wikia.com/fandom-ae-assets/prebid.js/2.44.0/20191223.min.js';
```
Comment (nandy-andy) [inline src/ad-engine/services/pbjs-factory.ts]: Missing `v` 🤦‍♂

File: platforms/futhead/bidders/prebid/nobid.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import { DeviceMode } from '@platforms/shared';
+
+export function getNoBidContext(device: DeviceMode): any {
+	const context = {
+		desktop: {
+			enabled: false,
+			slots: {
+				'01_LB': {
+					sizes: [[728, 90], [970, 250]],
+					siteId: '21872987314',
```
Comment (dmnsobczak) [inline platforms/futhead/bidders/prebid/nobid.ts]: For Futhead `siteId` should be `21874833814`

File: platforms/futhead/setup/context/prebid/futhead-prebid-config.setup.ts [hunk 3]
```diff
@@ -17,6 +18,7 @@ export class FutheadPrebidConfigSetup implements PrebidConfigSetup {
 		context.set('bidders.prebid.appnexus', getAppNexusContext(mode));
 		context.set('bidders.prebid.criteo', getCriteoContext(mode));
 		context.set('bidders.prebid.indexExchange', getIndexExchangeContext(mode));
+		context.set('bidders.prebid.noBid', getNoBidContext(mode));
```
Comment (dmnsobczak) [inline platforms/futhead/setup/context/prebid/futhead-prebid-config.setup.ts]: I guess Nobid is a one word so it should be `bidders.prebid.nobid` but who cares ¯\_(ツ)_/¯

File: src/ad-engine/services/pbjs-factory.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@ import { logger, scriptLoader } from '../utils';
 import { context } from './context-service';
 
 const prebidLibraryUrl =
-	'//origin-images.wikia.com/fandom-ae-assets/prebid.js/v2.44.0/20191213.min.js';
+	'//origin-images.wikia.com/fandom-ae-assets/prebid.js/v2.44.0/20191223.min.js';
```
Comment (dmnsobczak) [inline src/ad-engine/services/pbjs-factory.ts]: We need to clean this up with release in January.

File: platforms/futhead/setup/context/prebid/futhead-prebid-config.setup.ts (hunk 3 above)
Comment (nandy-andy) [reply platforms/futhead/setup/context/prebid/futhead-prebid-config.setup.ts]: The adapter in Prebid.js is called `nobidBidAdapter`, so you're right, I guess. I'll think about how impactful the change is and if it's small I'll change it 🙇

File: src/platforms-shared/bidders-state-setup.ts [hunk 5]
```diff
@@ -22,6 +22,7 @@ export function setupBidders(context: Context, instantConfig: InstantConfigServi
 		context.set('bidders.prebid.indexExchange.enabled', instantConfig.get('icPrebidIndexExchange'));
 		context.set('bidders.prebid.kargo.enabled', instantConfig.get('icPrebidKargo'));
 		context.set('bidders.prebid.lkqd.enabled', instantConfig.get('icPrebidLkqd'));
+		context.set('bidders.prebid.nobid.enabled', instantConfig.get('icPrebidNoBid'));
```
Comment (dmnsobczak) [inline src/platforms-shared/bidders-state-setup.ts]: `icPrebidNobid`? 🙈

File: src/ad-bidders/prebid/adapters/nobid.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import { PrebidAdapter } from '../prebid-adapter';
+import { PrebidAdSlotConfig } from '../prebid-models';
+
+export class NoBid extends PrebidAdapter {
```
Comment (dmnsobczak) [inline src/ad-bidders/prebid/adapters/nobid.ts]: `export class Nobid`? 🙈

File: src/platforms-shared/bidders-state-setup.ts (hunk 5 above)
Comment (nandy-andy) [reply src/platforms-shared/bidders-state-setup.ts]: I'd love to but it doesn't seem so easy. There is no EDIT button in ICBM I'd need to connect to MySQL and update the data directly in the database, wouldn't I?

Comment (dmnsobczak) [reply src/platforms-shared/bidders-state-setup.ts]: Oh, you're right. Let's ignore it then 🤷‍♂

Comment (nandy-andy) [reply src/platforms-shared/bidders-state-setup.ts]: Or... I can create a new ICBM var and we can remove the NoBid one in January ;) Better to do it now than later 👍

Comment (nandy-andy) [reply src/platforms-shared/bidders-state-setup.ts]: Oh, it's not that easy... ICBM isn't case-sensitive 🙊

**Pull request merged**